MbDotNet/Models/Stubs/HttpStub.cs

```diff
@@ -251,6 +251,20 @@ public HttpStub OnPathAndMethodEqual(string path, Method method)
             return On(predicate);
         }
 
+        /// <summary>
+        /// Adds a predicate to the stub that allows you to inject JavaScript to determine if the
+        /// predicate should match or not.
+        /// </summary>
+        /// <param name="javaScriptFunction">JavaScript function that accepts the request object (and
+        /// optionally a logger) and returns true or false</param>
+        /// <returns>The stub that the predicate was added to</returns>
+        public HttpStub OnJavaScriptFunction(string javaScriptFunction)
```

Comment:
Same with the methods on `HttpStub` and `TcpStub`, I would prefer `OnInjectedFunction` to make it clear that this is using the predicate injection feature.